src/components/WideButton/WideButton.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+
+import generateColors from '../../utils/generateColors';
+import {
+  BRAG_VARIANT,
+  CONFESS_VARIANT,
+  FRONT_END_VARIANT,
+  BACK_END_VARIANT,
+  DEV_OPS_VARIANT
+} from '../../constants/colourVariants';
+import Button from './Button';
+import LightbublIcon from './LightbulbIcon';
+import SaltGrignderIcon from './SaltGrinderIcon';
+import LongArrow from './LongArrowIcon';
+
+const ICONS = {
+  [BRAG_VARIANT]: <LightbublIcon />,
+  [CONFESS_VARIANT]: <SaltGrignderIcon />,
+  [FRONT_END_VARIANT]: <LongArrow />,
+  [BACK_END_VARIANT]: <LongArrow />,
+  [DEV_OPS_VARIANT]: <LongArrow />
+};
+
+function WideButton({ variant, children, className, to, onClick }) {
+  // Render link if 'to' prop is passed, button with handler otherwise
+  const props = {
+    ...(to ? { as: Link, to } : { onClick: onClick })
+  };
+
+  return (
+    <Button variant={generateColors(variant)} className={className} {...props}>
+      {children}
+      {ICONS[variant]}
```

Comment:
```
  const buttonIcon = ICONS[variant]
  return (
    <Button variant={generateColors(variant)} className={className} {...props}>
      {children}
      {<Icon />}
    </Button>
  );
```

Reply:
Also you are not passing `variant` to the `Button` but already generated color scheme. `variant` prop name is invalid.